cmsplugin_articles_ai/models.py

```diff
@@ -0,0 +1,289 @@
+        _("published until"), null=True, blank=True
+    )
+    highlight = models.BooleanField(
+        _("highlight"), default=False, help_text=_("Highlight as important.")
+    )
+    author = models.ForeignKey(User, verbose_name=_("author"))
+    tags = models.ManyToManyField(
+        Tag, verbose_name=_("tags"), related_name="articles", blank=True
+    )
+    main_image = FilerImageField(
+        verbose_name=_("main image"),
+        related_name="main_image",
+        null=True,
+        blank=True,
+    )
+    main_content = HTMLField(verbose_name=_("content"))
+
+    created_at = models.DateTimeField(_("creation time"), auto_now_add=True)
+    modified_at = models.DateTimeField(_("last modified"), auto_now=True)
+
+    objects = ArticleQuerySet.as_manager()
+
+    class Meta:
+        verbose_name = _("article")
+        verbose_name_plural = _("articles")
+        ordering = ('-published_from', '-pk')
+
+    def __str__(self):
+        return self.title
+
+    @property
+    def is_highlighted(self):
+        return self.highlight
+
+    @property
+    def is_public(self):
+        now = timezone.now()
+        if not self.published_from:
+            return False
+        if self.published_from > now:
+            return False
+        if self.published_until and self.published_until < now:
+            return False
+        return True
+
+    def get_absolute_url(self):
+        """
+        Returns URL to the articles detail page.
+        Note that this will raise NoReverseMatch if articles app hasn't
+        been hooked to any CMS page.
+        """
+        return reverse("article", kwargs={"slug": self.slug})
+
+
+class ArticleQuerySet(models.QuerySet):
+
+    def images(self):
+        return self.filter(is_image=True)
+
+    def non_images(self):
+        return self.filter(is_image=False)
+
+
+class ArticleAttachment(models.Model):
+    name = models.CharField(_("name"), max_length=64, blank=True)
+    attachment_file = FilerFileField(verbose_name=_("file"))
+    article = models.ForeignKey(
+        Article, verbose_name=_("article"), related_name="attachments"
+    )
+    is_image = models.BooleanField(default=False, editable=False, db_index=True)
+
+    objects = ArticleQuerySet.as_manager()
+
+    class Meta:
+        verbose_name = _("article attachment")
+        verbose_name_plural = _("article attachments")
+
+    def __str__(self):
+        return "%s (%s)" % (self.name, self.filename)
+
+    @property
+    def icon_url(self):
+        obj = self.attachment_file.get_real_instance()
+        return obj.icons.get("16", "")
+
+    @property
+    def title(self):
+        return self.name or self.filename
+
+    @property
+    def url(self):
+        return self.attachment_file.url
+
+    @property
+    def size(self):
+        return self.attachment_file.size
+
+    @property
+    def filename(self):
+        return self.attachment_file.original_filename
+
+    def is_file_image(self):
```

Comment:
Prefix this method's name with an underscore; the primary API to find out whether a file is an image or not is `is_image`, so this shouldn't be used by external callers.